### PR TITLE
Show more context on settings menu

### PR DESCRIPTION
**Pull request by zisra**
Screenshot 2023-05-25 at 11 24 00 PM

Captions and Playback speed now show more info. I believe this is important because it actually makes the user experience simpler. Users would have to click less buttons to see what current subtitle they are using. I implemented this because I felt like I needed it personally, so maybe others will too.

### CODE REVIEW

**Comment by frost768** [comment]
Never felt the need for it until I saw it and now I think it is a must

File: src/video/components/actions/list-entries/CaptionsSelectionAction.tsx [hunk 1]
```diff
@@ -1,18 +1,52 @@
...
+  );
+  const captionId = source.source?.caption?.id;
+  const captionName =
+    linkedCaptions.find((caption) => caption.id === captionId)?.langIso ??
+    (captionId === "external-custom" ? "Custom" : null);
+
+  return (
+    <div className="rounded-md bg-denim-300 px-2 py-1 transition-colors">
+      <p className="text-center text-xs font-bold text-slate-300 transition-colors">
+        {captionName ?? "None"}
```
Comment (JamesHawkinss) [inline src/video/components/actions/list-entries/CaptionsSelectionAction.tsx:34]: I think if there are no captions, there should be no additional information displayed. Only show the language if there is a language to show.

File: src/video/components/actions/list-entries/PlaybackSpeedSelectionAction.tsx [hunk 2]
```diff
@@ -1,18 +1,40 @@
 import { useTranslation } from "react-i18next";
 
 import { Icons } from "@/components/Icon";
+import { useVideoPlayerDescriptor } from "@/video/state/hooks";
+import { useMediaPlaying } from "@/video/state/logic/mediaplaying";
+import { useMeta } from "@/video/state/logic/meta";
```
Comment (JamesHawkinss) [inline src/video/components/actions/list-entries/PlaybackSpeedSelectionAction.tsx:6]: See the linting errors here, 

'useMeta' is defined but never used
'useSource' is defined but never used

Will fail to lint, please fix!

File: src/video/components/actions/list-entries/QualityDisplayAction.tsx [hunk 3]
```diff
@@ -1,17 +1 @@
-import { useVideoPlayerDescriptor } from "@/video/state/hooks";
-import { useSource } from "@/video/state/logic/source";
```
Comment (JamesHawkinss) [inline src/video/components/actions/list-entries/QualityDisplayAction.tsx:1]: This file still exists, is causing more lint errors

**Review by JamesHawkinss — CHANGES_REQUESTED** [review]

**Comment by zisra** [comment]
Before I make this ready to merge, there are two alternatives that I have in mind:
- Keeping the chevron on all items
- Showing the current source for source (Because the source is not 720p, it's SuperStream) and the quality for download

File: src/video/components/actions/list-entries/DownloadAction.tsx [hunk 4]
```diff
@@ -9,6 +9,21 @@ import { useSource } from "@/video/state/logic/source";
...
+export function CurrentQuality() {
+  const descriptor = useVideoPlayerDescriptor();
+  const source = useSource(descriptor);
+
+  if (!source.source) return null;
+
+  return (
+    <div className="rounded-md bg-denim-300 px-2 py-1 transition-colors">
+      <p className="text-center text-xs font-bold text-slate-300 transition-colors">
+        {source.source.quality}
```
Comment (JamesHawkinss) [inline src/video/components/actions/list-entries/DownloadAction.tsx:21]: Consider hiding quality if it's of type UNKNOWN

**Comment by zisra** [comment]
@JipFr Done. Looks like this now:
Screenshot 2023-05-27 at 6 13 59 PM
I also added support to download m3u8 files if that's ok. It's useful for me because I can use ffmpeg to later convert

**Comment by genolith** [comment]
Can you add a option to choose video quality?

File: src/video/components/actions/list-entries/CaptionsSelectionAction.tsx [hunk 5]
```diff
@@ -9,10 +14,38 @@ interface Props {
...
+      right={
+        captionName ? (
+          <div className="rounded-md bg-denim-300 px-2 py-1 transition-colors">
+            <p className="text-center text-xs font-bold text-slate-300 transition-colors">
+              {captionName}
+            </p>
+          </div>
+        ) : null
+      }
+      noChevron={!!captionName} // TODO: Make this false if <CurrentCaption /> returns null
```
Comment (JamesHawkinss) [inline src/video/components/actions/list-entries/CaptionsSelectionAction.tsx:47]: Complete the TODO please

File: src/video/components/actions/list-entries/DownloadAction.tsx [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import { useTranslation } from "react-i18next";
 
-import { MWStreamType } from "@/backend/helpers/streams";
+import { MWStreamQuality, MWStreamType } from "@/backend/helpers/streams";
```
Comment (JamesHawkinss) [inline src/video/components/actions/list-entries/DownloadAction.tsx:3]: 'MWStreamType' is defined but never used

**Review by JamesHawkinss — CHANGES_REQUESTED** [review]
Have you checked that the design is fine on mobile?

**Comment by binaryoverload** [comment]
Since this will be re-designed, we'll be shutting this PR as it won't be able to be used in v4.

**Pull request closed without merging**